src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -1076,6 +1099,7 @@ private AdjustmentRule GetAdjustmentRuleForTime(DateTime dateTime, bool dateTime
                 int compareResult = CompareAdjustmentRuleToDateTime(rule, previousRule, dateTime, date, dateTimeisUtc);
                 if (compareResult == 0)
                 {
+                    ruleIndex = median;
```

Comment:
> ruleIndex = median; [](start = 19, length = 20)

what you think if we added the index inside the rule itself? and we don't have to pass it around with the rule.